hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/SimpleCopyListing.java

```diff
@@ -316,6 +303,42 @@ protected void doBuildListingWithSnapshotDiff(
     }
   }
 
+  /**
+   * Handle create Diffs and add to the copyList.
+   * If the path is a directory, iterate it recursively and add the paths
+   * to the result copyList.
+   *
+   * @param fileListWriter the list for holding processed results
+   * @param context The DistCp context with associated input options
+   * @param sourceRoot The rootDir of the source snapshot
+   * @param sourceFS the source Filesystem
+   * @param fileStatuses store the result fileStatuses to add to the copyList
+   * @param diff the SnapshotDiff report
+   * @throws IOException
+   */
+  protected void addCreateDiffsToFileListing(SequenceFile.Writer fileListWriter,
+      DistCpContext context, Path sourceRoot, FileSystem sourceFS,
+      List<FileStatusInfo> fileStatuses, DiffInfo diff) throws IOException {
+    addToFileListing(fileListWriter, sourceRoot, diff.getTarget(), context);
+
+    FileStatus sourceStatus = sourceFS.getFileStatus(diff.getTarget());
```

Comment:
Have you thought about just having a advanced flag to control this? I am not sure we will be having many implementations of these copyListings 
I am not against the current design, but it's a simple thought to keep it simple

Reply:
Yes, I didn't do so to change existing behaviour of SimpleCopyListing.

Reply:
I have now changed it to use a flag now.